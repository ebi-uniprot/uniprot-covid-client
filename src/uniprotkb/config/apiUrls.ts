import queryString from 'query-string';
import joinUrl from 'url-join';
import {
  getApiSortDirection,
  SortDirection,
  SelectedFacet,
  FileFormat,
  fileFormatsWithColumns,
  fileFormatToUrlParameter,
} from '../types/resultsTypes';
import { SortableColumn } from '../types/columnTypes';

<<<<<<< HEAD:src/utils/apiUrls.ts
export const joinUrl = (...args: string[]) => urlJoin(args);

const devPrefix = 'https://wwwdev.ebi.ac.uk';
const prodPrefix = 'https://www.ebi.ac.uk';
const covidPrefix = 'https://www.ebi.ac.uk/uniprot/api/covid-19';
=======
export const devPrefix = 'https://wwwdev.ebi.ac.uk';
export const prodPrefix = 'https://www.ebi.ac.uk';
>>>>>>> uniprot/2020_05__better_fetch_and_sequences:src/uniprotkb/config/apiUrls.ts

const apiUrls = {
  // uniprotkb advanced search terms
  advancedSearchTerms: joinUrl(
    devPrefix,
    '/uniprot/api/configure/uniprotkb/search_terms'
  ),
  // Annotation evidence used by advanced search
  evidences: {
    annotation: joinUrl(
      devPrefix,
      '/uniprot/api/configure/uniprotkb/annotation_evidences'
    ),
    // Go evidences used by advanced go search
    // "itemType": "goterm",
    go: joinUrl(devPrefix, '/uniprot/api/configure/uniprotkb/go_evidences'),
  },
  // Database cross references used by advanced search

  databaseXrefs: joinUrl(
    devPrefix,
    '/uniprot/api/configure/uniprotkb/databases'
  ),
  // Database cross reference fields in result column configure
  // "itemType": "database",
  databaseFields: joinUrl(
    devPrefix,
    '/uniprot/api/configure/uniprotkb/databasefields'
  ),
  // All result fields except database cross reference fields
  resultsFields: joinUrl(
    devPrefix,
    '/uniprot/api/configure/uniprotkb/resultfields'
  ),
  // Retrieve results
  search: joinUrl(covidPrefix, '/uniprotkb/search'),
  download: joinUrl(covidPrefix, '/uniprotkb/download'),
  variation: joinUrl(prodPrefix, '/proteins/api/variation'),

  entry: (accession: string) =>
<<<<<<< HEAD:src/utils/apiUrls.ts
    joinUrl(covidPrefix, '/uniprotkb/accession', accession),

=======
    joinUrl(devPrefix, '/uniprot/api/uniprotkb/accession', accession),
>>>>>>> uniprot/2020_05__better_fetch_and_sequences:src/uniprotkb/config/apiUrls.ts
  sequenceFasta: (accession: string) => `${apiUrls.entry(accession)}.fasta`,
  entryDownload: (accession: string, format: FileFormat) =>
    format === FileFormat.fastaCanonicalIsoform
      ? `${apiUrls.search}?${queryString.stringify({
          query: `accession:${accession}`,
          includeIsoform: true,
          format: fileFormatToUrlParameter.get(
            FileFormat.fastaCanonicalIsoform
          ),
        })}`
      : `${apiUrls.entry(accession)}.${fileFormatToUrlParameter.get(format)}`,
  entryPublications: (accession: string) =>
    joinUrl(covidPrefix, '/uniprotkb/accession', accession, '/publications'),
};

export default apiUrls;

const RE_QUERY = /\?$/;
export const getSuggesterUrl = (url: string, value: string) =>
  joinUrl(devPrefix, url.replace(RE_QUERY, value));

export const createFacetsQueryString = (facets: SelectedFacet[]) =>
  /**
   * Add double quotes to facet values which contain
   * spaces as otherwise the backend doesn't escape special characters
   * such as '.' or '-'.
   * Single word values shouldn't have double quotes as they can be boolean.
   * Range queries (/^\[.*]$/) should not have double quotes either.
   * */
  facets.reduce(
    (queryAccumulator, facet) =>
      `${queryAccumulator} AND (${facet.name}:${
        facet.value.indexOf(' ') >= 0 && !facet.value.match(/^\[.*\]$/)
          ? `"${facet.value}"`
          : facet.value
      })`,
    ''
  );

export const createAccessionsQueryString = (accessions: string[]) =>
  accessions.map((accession) => `accession:${accession}`).join(' OR ');

export const getAPIQueryUrl = (
  query: string,
  columns: string[] | null,
  selectedFacets: SelectedFacet[],
  sortColumn: SortableColumn | undefined = undefined,
  sortDirection: SortDirection | undefined = SortDirection.ascend
) =>
  `${apiUrls.search}?${queryString.stringify({
    query: `${query}${createFacetsQueryString(selectedFacets)}`,
    fields: columns && columns.join(','),
    facets:
<<<<<<< HEAD:src/utils/apiUrls.ts
      'reviewed,popular_organism,other_organism,proteins_with,existence,annotation_score',
=======
      'reviewed,model_organism,proteins_with,existence,annotation_score,length',
>>>>>>> uniprot/2020_05__better_fetch_and_sequences:src/uniprotkb/config/apiUrls.ts
    sort:
      sortColumn &&
      `${sortColumn} ${getApiSortDirection(SortDirection[sortDirection])}`,
  })}`;

export const getUniProtPublicationsQueryUrl = (
  accession: string,
  selectedFacets: SelectedFacet[]
) =>
  `${apiUrls.entryPublications(accession)}?${queryString.stringify({
    facets: 'source,category,study_type',
    query: selectedFacets
      .map((facet) => `(${facet.name}:"${facet.value}")`)
      .join(' AND '),
  })}`;

export const getDownloadUrl = ({
  query,
  columns,
  selectedFacets,
  sortColumn,
  sortDirection = SortDirection.ascend,
  fileFormat,
  compressed = false,
  size,
  selectedAccessions = [],
}: {
  query: string;
  columns: string[];
  selectedFacets: SelectedFacet[];
  sortColumn: SortableColumn;
  sortDirection: SortDirection;
  fileFormat: FileFormat;
  compressed: boolean;
  size?: number;
  selectedAccessions: string[];
}) => {
  const parameters: {
    query: string;
    format: string;
    fields?: string;
    sort?: string;
    includeIsoform?: boolean;
    size?: number;
    compressed?: boolean;
  } = {
    query: selectedAccessions.length
      ? createAccessionsQueryString(selectedAccessions)
      : `${query}${createFacetsQueryString(selectedFacets)}`,
    // fallback to json if something goes wrong
    format: fileFormatToUrlParameter.get(fileFormat) || 'json',
  };
  const isColumnFileFormat = fileFormatsWithColumns.includes(fileFormat);
  if (isColumnFileFormat && sortColumn) {
    parameters.sort = `${sortColumn} ${getApiSortDirection(
      SortDirection[sortDirection]
    )}`;
  }
  if (fileFormat === FileFormat.fastaCanonicalIsoform) {
    parameters.includeIsoform = true;
  }
  if (isColumnFileFormat && columns) {
    parameters.fields = columns.join(',');
  }
  if (size && !selectedAccessions.length) {
    parameters.size = size;
  }
  if (compressed) {
    parameters.compressed = true;
  }
  return `${apiUrls.download}?${queryString.stringify(parameters)}`;
};

<<<<<<< HEAD:src/utils/apiUrls.ts
export const urlsAreEqual = (
  url1: string,
  url2: string,
  ignoreParams: string[] = []
) => {
  const urlObject1 = queryString.parseUrl(url1);
  const urlObject2 = queryString.parseUrl(url2);
  if (urlObject1.url !== urlObject2.url) {
    return false;
  }
  const paramsIntersection = uniq([
    ...Object.keys(urlObject1.query),
    ...Object.keys(urlObject1.query),
  ]);
  return paramsIntersection.every(
    (param) =>
      ignoreParams.includes(param) ||
      urlObject1.query[param] === urlObject2.query[param]
  );
=======
export const literatureApiUrls = {
  literature: joinUrl(devPrefix, '/uniprot/api/literature'),
>>>>>>> uniprot/2020_05__better_fetch_and_sequences:src/uniprotkb/config/apiUrls.ts
};

export const getPublicationURL = (id: string) =>
  joinUrl(literatureApiUrls.literature, id);

export const getPublicationsURL = (ids: string[]) =>
  `${literatureApiUrls.literature}/search?query=(${ids
    .map((id) => `id:${id}`)
    .join(' OR ')})`;
