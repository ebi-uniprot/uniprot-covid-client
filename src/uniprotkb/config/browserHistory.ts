import { createBrowserHistory } from 'history';

declare const BASE_URL: string;

export default createBrowserHistory({ basename: BASE_URL });
