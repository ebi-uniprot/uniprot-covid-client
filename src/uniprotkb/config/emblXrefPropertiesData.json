{
  "Genomic_DNA": "Genomic DNA Translation",
  "Genomic_RNA": "Genomic RNA Translation",
  "Other_DNA": "Other DNA Translation",
  "Other_RNA": "Other RNA Translation",
  "Transcribed_RNA": "Transcribed RNA Translation",
  "Unassigned_DNA": "Unassigned DNA Translation",
  "Unassigned_RNA": "Unassigned RNA Translation",
  "Viral_cRNA": "Viral cRNA Translation",
  "mRNA": "mRNA Translation",
  "ALT_FRAME": "Frameshift",
  "ALT_INIT": "Different initiation",
  "ALT_SEQ": "Sequence problems.",
  "ALT_TERM": "Different termination.",
  "NOT_ANNOTATED_CDS": "No translation available."
}
