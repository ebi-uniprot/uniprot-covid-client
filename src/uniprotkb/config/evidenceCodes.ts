import { Evidence } from '../types/modelTypes';

export enum ECO {
  EXP = 269,
  HTP = 6056,
  IDA = 314,
  HDA = 7005,
  IPI = 353,
  IMP = 315,
  HMP = 7001,
  IGI = 316,
  HGI = 7003,
  IEP = 270,
  HEP = 7007,
  ISS = 250,
  ISO = 266,
  ISA = 247,
  ISM = 255,
  IGC = 317,
  IBA = 318,
  IBD = 319,
  IKR = 320,
  IRD = 321,
  RCA = 245,
  TAS = 304,
  NAS = 303,
  IC = 305,
  ND = 307,
  IEA = 501,
  MI = 312,
  AI = 313,
  AA = 256,
  MIXM = 244,
  MIXA = 213,
  SGNM = 260,
  SGNA = 259,
}

enum labels {
  IMPORTED = 'Imported',
  COMBINED = 'Combined sources',
  INTERPRO = 'InterPro annotation',
  SIMILARITY = 'By similarity',
  CURATED = 'Curated',
  PUBLICATION = 'publication',
  AA = 'automatic annotation',
  SEQ_ANA = 'Sequence analysis',
}

export type EvidenceData = {
  manual: boolean;
  label: string;
  description: string;
  labelRender?: (evidences: Evidence[]) => string;
};

const publicationCountRenderer = (evidences: Evidence[]) => {
  const { length } = evidences;
  return length > 0
    ? `${length} ${labels.PUBLICATION}${length > 1 ? 's' : ''}`
    : labels.CURATED;
};

const rulesCountRenderer = (evidences: Evidence[]) => {
  const { length } = evidences;
  const isSAMPhobius = evidences.some(
    (evidence) =>
      typeof evidence.source !== 'undefined' &&
      evidence.source === 'SAM' &&
      evidence.id === 'Phobius'
  );
  if (isSAMPhobius) {
    return labels.SEQ_ANA;
  }
  return `${length} ${labels.AA}${length > 1 ? 's' : ''}`;
};

export const getEvidenceCodeData = (eco: string): EvidenceData | null => {
  const num = Number(eco.split(':')[1]);
  switch (num) {
    case ECO.EXP:
      return {
        manual: true,
        label: 'Manual assertion based on experiment',
        description: 'Inferred from experiment',
        labelRender: publicationCountRenderer,
      };
    case ECO.HTP:
      return {
        manual: true,
        label: 'Manual assertion based on experiment',
        description: 'Inferred from high throughput experiment',
      };
    case ECO.IDA:
      return {
        manual: true,
        label: 'Manual assertion based on experiment',
        description: 'Inferred from direct assay',
      };
    case ECO.HDA:
      return {
        manual: true,
        label: 'Manual assertion based on experiment',
        description: 'Inferred from high throughput direct assay',
      };
    case ECO.IPI:
      return {
        manual: true,
        label: 'Manual assertion based on experiment',
        description: 'Inferred from physical interaction',
      };
    case ECO.IMP:
      return {
        manual: true,
        label: 'Manual assertion based on experiment',
        description: 'Inferred from mutant phenotype',
      };
    case ECO.HMP:
      return {
        manual: true,
        label: 'Manual assertion based on experiment',
        description: 'Inferred from high throughput mutant phenotype',
      };
    case ECO.IGI:
      return {
        manual: true,
        label: 'Manual assertion based on experiment',
        description: 'Inferred from genetic interaction',
      };
    case ECO.HGI:
      return {
        manual: true,
        label: 'Manual assertion based on experiment',
        description: 'Inferred from high throughput genetic interaction',
      };
    case ECO.IEP:
      return {
        manual: true,
        label: 'Manual assertion based on experiment',
        description: 'Inferred from expression pattern',
      };
    case ECO.HEP:
      return {
        manual: true,
        label: 'Manual assertion based on experiment',
        description: 'Inferred from high throughput expression pattern',
      };
    case ECO.ISS:
      return {
        manual: true,
        label: 'Manual assertion inferred from sequence similarity',
        description: 'Inferred from sequence or structural similarity',
        labelRender: () => labels.SIMILARITY,
      };
    case ECO.ISO:
      return {
        manual: true,
        label: 'Manual assertion based on experiment',
        description: 'Inferred from sequence orthology',
      };
    case ECO.ISA:
      return {
        manual: true,
        label: 'Manual assertion based on experiment',
        description: 'Inferred from sequence alignment',
      };
    case ECO.ISM:
      return {
        manual: true,
        label: 'Automatic assertion according to rules',
        description: 'Inferred from sequence model',
        labelRender: rulesCountRenderer,
      };
    case ECO.IGC:
      return {
        manual: true,
        label: 'Manual assertion based on experiment',
        description: 'Inferred from genomic context',
      };
    case ECO.IBA:
      return {
        manual: true,
        label: 'Manual assertion based on experiment',
        description: 'Inferred from biological aspect of ancestor',
      };
    case ECO.IBD:
      return {
        manual: true,
        label: 'Manual assertion based on experiment',
        description: 'Inferred from biological aspect of descendant',
      };
    case ECO.IKR:
      return {
        manual: true,
        label: 'Manual assertion based on experiment',
        description: 'Inferred from key residues',
      };
    case ECO.IRD:
      return {
        manual: true,
        label: 'Manual assertion based on experiment',
        description: 'Inferred from rapid divergence',
      };
    case ECO.RCA:
      return {
        manual: true,
        label: 'Manual assertion based on experiment',
        description: 'Inferred from reviewed computational analysis',
      };
    case ECO.TAS:
      return {
        manual: true,
        label: 'Manual assertion based on experiment',
        description: 'Traceable author statement',
      };
    case ECO.NAS:
      return {
        manual: true,
        label: 'Manual assertion based on opinion',
        description: 'Non-traceable author statement',
        labelRender: publicationCountRenderer,
      };
    case ECO.IC:
      return {
        manual: true,
        label: 'Manual assertion inferred from experiment',
        description: 'Inferred by curator',
        labelRender: (evidences: Evidence[]) => {
          return evidences.some((evidence) => evidence.source)
            ? publicationCountRenderer(evidences)
            : 'Curated';
        },
      };
    case ECO.ND:
      return {
        manual: true,
        label: 'Manual assertion based on experiment',
        description: 'No biological data available',
      };
    case ECO.IEA:
      return {
        manual: false,
        label: 'Manual assertion based on experiment',
        description: 'Inferred from electronic annotation',
      };
    case ECO.MI:
      return {
        manual: true,
        label: 'Manual assertion inferred from database entries',
        description: 'Manually imported',
        labelRender: () => labels.IMPORTED,
      };
    case ECO.AI:
      return {
        manual: false,
        label: 'Automatic assertion inferred from database entries',
        description: 'Automatically imported',
        labelRender: () => labels.IMPORTED,
      };
    case ECO.AA:
      return {
        manual: false,
        label: 'Automatic assertion according to rules',
        description: 'Automatically inferred from sequence model',
        labelRender: rulesCountRenderer,
      };
    case ECO.MIXM:
      return {
        manual: true,
        label:
          'Manual assertion inferred from combination of experimental and computational evidence',
        description: 'Combinatorial evidence used in manual assertion',
        labelRender: () => labels.COMBINED,
      };
    case ECO.MIXA:
      return {
        manual: false,
        label:
          'Automatic assertion inferred from combination of experimental and computational evidence',
        description: 'Combinatorial evidence used in automatic assertion',
        labelRender: () => labels.COMBINED,
      };
    case ECO.SGNM:
      return {
        manual: true,
        label: 'Manual assertion inferred from signature match',
        description:
          'Match to InterPro member signature evidence used in manual assertion',
      };
    case ECO.SGNA:
      return {
        manual: false,
        label: 'Automatic assertion inferred from signature match',
        description:
          'Match to InterPro member signature evidence used in automatic assertion',
        labelRender: () => labels.INTERPRO,
      };
    default:
      return null;
  }
};
