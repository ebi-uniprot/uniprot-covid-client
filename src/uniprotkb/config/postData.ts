import axios from 'axios';

export default function postData(
  url: string,
  options: {
    headers: Record<string, string>;
    data?: FormData;
  } = {
    headers: {},
  }
) {
  const { headers = {}, data } = options;
  return axios.post(url, data, {
    headers: {
      Accept: 'application/json',
      ...headers,
    },
  });
}
