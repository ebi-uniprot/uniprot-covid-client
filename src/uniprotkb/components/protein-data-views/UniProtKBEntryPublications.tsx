import React, { FC, Fragment } from 'react';
import { Publication, Loader, Message } from 'franklin-sites';
import useDataApi from '../../../shared/hooks/useDataApi';
import { getPublicationsURL } from '../../config/apiUrls';
import { MessageLevel } from '../../../messages/types/messagesTypes';
import { LiteratureAPI } from '../../types/literatureTypes';
import formatCitationData from '../../adapters/literatureConverter';

const UniProtKBEntryPublications: FC<{
  pubmedIds: string[];
}> = ({ pubmedIds }) => {
  const url = getPublicationsURL(pubmedIds);
  const { loading, data, status, error } = useDataApi(url);

  if (error) {
    return (
      <Message level={MessageLevel.FAILURE}>
        {status}: {error.message}
      </Message>
    );
  }

  if (loading || !data) {
    return <Loader />;
  }

  const { results }: { results: LiteratureAPI[] } = data;
  return (
    <Fragment>
      {results &&
        results
          .map(literatureItem => ({
            ...literatureItem,
            ...formatCitationData(literatureItem.citation),
          }))
          .map(({ citation, statistics, pubmedId, journalInfo }) => (
            <Publication
              title={citation.title}
              authors={citation.authors}
              key={`${citation.title}-${citation.citationType}-${citation.journal}`}
              pubmedId={pubmedId}
              statistics={statistics}
              journalInfo={journalInfo}
            />
          ))}
    </Fragment>
  );
};

export default UniProtKBEntryPublications;
