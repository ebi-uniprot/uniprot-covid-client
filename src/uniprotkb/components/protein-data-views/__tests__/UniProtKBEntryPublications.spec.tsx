import React from 'react';
import UniProtKBEntryPublications from '../UniProtKBEntryPublications';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import { getPublicationsURL } from '../../../config/apiUrls';
import mockUniProtKBEntryPublications from './__mocks__/uniprotKBEntryPublications.json';

import useDataApi from '../../../../shared/hooks/useDataApi';
jest.mock('../../../../shared/hooks/useDataApi', () => jest.fn());

describe('UniProtKBEntryPublications', () => {
  it('Should make a call with pubmed ids and render properly', () => {
    useDataApi.mockImplementation(() => ({
      loading: false,
      data: mockUniProtKBEntryPublications,
    }));
    const pubMedIds = ['123', '456'];
    const { asFragment } = render(
      <Router>
        <UniProtKBEntryPublications pubmedIds={pubMedIds} />
      </Router>
    );
    expect(useDataApi).toHaveBeenCalledWith(getPublicationsURL(pubMedIds));
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should make a call with pubmed ids', () => {
    useDataApi.mockImplementation(() => ({
      error: {
        message: 'There has been an error',
      },
    }));
    const pubMedIds = ['123', '456'];
    const { asFragment } = render(
      <Router>
        <UniProtKBEntryPublications pubmedIds={pubMedIds} />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
