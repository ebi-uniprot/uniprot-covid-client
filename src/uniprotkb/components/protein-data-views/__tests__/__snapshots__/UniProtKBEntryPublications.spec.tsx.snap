// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UniProtKBEntryPublications Should make a call with pubmed ids 1`] = `
<DocumentFragment>
  <div
    class="message message--failure"
    role="status"
  >
    <section
      class="message__content"
    >
      <svg
        fill="currentColor"
        height="1.125em"
        viewBox="0 0 18 18"
        width="1.125em"
      >
        <path
          clip-rule="evenodd"
          d="M18 9A9 9 0 110 9a9 9 0 0118 0zM8 4a1 1 0 112 0v6a1 1 0 11-2 0V4zm1 9a1 1 0 100 2 1 1 0 000-2z"
          fill-rule="evenodd"
        />
      </svg>
      <small>
        : There has been an error
      </small>
    </section>
  </div>
</DocumentFragment>
`;

exports[`UniProtKBEntryPublications Should make a call with pubmed ids and render properly 1`] = `
<DocumentFragment>
  <section
    class="publication"
  >
    <section
      class="publication__columns"
    >
      <section
        class="publication__columns__main"
      >
        <h5>
          A peptide zipcode sufficient for anterograde transport within amyloid precursor protein.
        </h5>
        <section
          class="publication__authors"
        >
          <a
            href="/"
          >
            Satpute-Krishnan P.
          </a>
          , 
          <a
            href="/"
          >
            DeGiorgis J.A.
          </a>
          , 
          <a
            href="/"
          >
            Conley M.P.
          </a>
          , 
          <a
            href="/"
          >
            Jang M.
          </a>
          , 
          <a
            href="/"
          >
            Bearer E.L.
          </a>
        </section>
      </section>
      <section
        class="publication__columns__side"
      >
        <section
          class="publication__columns__side__item"
        >
          <svg
            fill="currentColor"
            height="2em"
            viewBox="0 0 30 32"
            width="1.875em"
          >
            <path
              d="M26.5 13.5h-8v8h8v-8z"
            />
            <path
              d="M26 14v7h-7v-7h7zm1-1h-9v9h9v-9z"
            />
            <path
              d="M29 1v28.06H17.05a3 3 0 00-2 .74V1.94a1 1 0 011-.94H29zm1-1H16.05a2 2 0 00-2 1.94V32h1a1.9 1.9 0 01.59-1.38 2 2 0 011.4-.56H30V0z"
            />
            <path
              d="M13.12 1a.91.91 0 01.88.94v27.78a2.78 2.78 0 00-1.81-.66H1V1h12.12zm0-1H0v30.06h12.19a1.84 1.84 0 011.32.56 1.93 1.93 0 01.55 1.38H15V1.94A1.91 1.91 0 0013.12 0z"
            />
            <path
              d="M22 5.5h5m-24 8h9m-9-4h7m-7-4h5m10 4h9m-24 8h9m-9 4h9m-9 4h9m6 0h9"
              stroke="currentColor"
              stroke-miterlimit="10"
            />
          </svg>
          <ul
            class="no-bullet"
          >
            <li>
              <a
                href="//www.ncbi.nlm.nih.gov/pubmed/17062754"
                rel="noopener noreferrer"
                target="_blank"
              >
                PubMed
              </a>
            </li>
            <li>
              <a
                href="//europepmc.org/article/MED/17062754"
                rel="noopener noreferrer"
                target="_blank"
              >
                Europe PMC
              </a>
            </li>
            <li>
              <small>
                <a
                  href="//dx.doi.org/10.1073/pnas.0607527103"
                >
                  Proc. Natl. Acad. Sci. U.S.A. 103:16532-16537(2006)
                </a>
              </small>
            </li>
          </ul>
        </section>
        <section
          class="publication__columns__side__item"
        >
          <section
            class="publication__statistics"
          >
            <section
              class="publication__statistics__item"
            >
              <a
                href="/"
              >
                <section>
                  <small>
                    Cited in
                  </small>
                </section>
                <section
                  class="publication__statistics__bubble"
                >
                  <span
                    class="bubble--small colour-pastel-blue"
                  >
                    2
                  </span>
                  <svg
                    fill="currentColor"
                    height="15"
                    viewBox="0 0 30 22"
                    width="15"
                  >
                    <path
                      d="M13.65 6.68V7.7a.14.14 0 010 .11c-.13 1.42-.47 2.81-1.01 4.13a15.73 15.73 0 01-2.7 4.54 15.87 15.87 0 01-3.96 3.4c-1.45.86-3 1.53-4.63 1.98 0 0 2.84-4.15 5.8-8.2h-.3a6.82 6.82 0 116.8-6.98zm16.35 0v1.13c-.13 1.42-.48 2.81-1.02 4.13a15.73 15.73 0 01-2.68 4.56 15.86 15.86 0 01-3.96 3.33 18.29 18.29 0 01-4.68 1.99s2.83-4.13 5.79-8.18h-.3A6.82 6.82 0 1130 6.68z"
                    />
                  </svg>
                </section>
              </a>
            </section>
          </section>
        </section>
      </section>
    </section>
  </section>
  <section
    class="publication"
  >
    <section
      class="publication__columns"
    >
      <section
        class="publication__columns__main"
      >
        <h5>
          Quantitative measurements and modeling of cargo-motor interactions during fast transport in the living axon.
        </h5>
        <section
          class="publication__authors"
        >
          <a
            href="/"
          >
            Seamster P.E.
          </a>
          , 
          <a
            href="/"
          >
            Loewenberg M.
          </a>
          , 
          <a
            href="/"
          >
            Pascal J.
          </a>
          , 
          <a
            href="/"
          >
            Chauviere A.
          </a>
          , 
          <a
            href="/"
          >
            Gonzales A.
          </a>
          , 
          <a
            href="/"
          >
            Cristini V.
          </a>
          , 
          <a
            href="/"
          >
            Bearer E.L.
          </a>
        </section>
      </section>
      <section
        class="publication__columns__side"
      >
        <section
          class="publication__columns__side__item"
        >
          <svg
            fill="currentColor"
            height="2em"
            viewBox="0 0 30 32"
            width="1.875em"
          >
            <path
              d="M26.5 13.5h-8v8h8v-8z"
            />
            <path
              d="M26 14v7h-7v-7h7zm1-1h-9v9h9v-9z"
            />
            <path
              d="M29 1v28.06H17.05a3 3 0 00-2 .74V1.94a1 1 0 011-.94H29zm1-1H16.05a2 2 0 00-2 1.94V32h1a1.9 1.9 0 01.59-1.38 2 2 0 011.4-.56H30V0z"
            />
            <path
              d="M13.12 1a.91.91 0 01.88.94v27.78a2.78 2.78 0 00-1.81-.66H1V1h12.12zm0-1H0v30.06h12.19a1.84 1.84 0 011.32.56 1.93 1.93 0 01.55 1.38H15V1.94A1.91 1.91 0 0013.12 0z"
            />
            <path
              d="M22 5.5h5m-24 8h9m-9-4h7m-7-4h5m10 4h9m-24 8h9m-9 4h9m-9 4h9m6 0h9"
              stroke="currentColor"
              stroke-miterlimit="10"
            />
          </svg>
          <ul
            class="no-bullet"
          >
            <li>
              <a
                href="//www.ncbi.nlm.nih.gov/pubmed/23011729"
                rel="noopener noreferrer"
                target="_blank"
              >
                PubMed
              </a>
            </li>
            <li>
              <a
                href="//europepmc.org/article/MED/23011729"
                rel="noopener noreferrer"
                target="_blank"
              >
                Europe PMC
              </a>
            </li>
            <li>
              <small>
                <a
                  href="//dx.doi.org/10.1088/1478-3975/9/5/055005"
                >
                  Phys. Biol. 9:055005-055005(2012)
                </a>
              </small>
            </li>
          </ul>
        </section>
        <section
          class="publication__columns__side__item"
        >
          <section
            class="publication__statistics"
          >
            <section
              class="publication__statistics__item"
            >
              <a
                href="/"
              >
                <section>
                  <small>
                    Cited in
                  </small>
                </section>
                <section
                  class="publication__statistics__bubble"
                >
                  <span
                    class="bubble--small colour-pastel-blue"
                  >
                    6
                  </span>
                  <svg
                    fill="currentColor"
                    height="15"
                    viewBox="0 0 30 22"
                    width="15"
                  >
                    <path
                      d="M13.65 6.68V7.7a.14.14 0 010 .11c-.13 1.42-.47 2.81-1.01 4.13a15.73 15.73 0 01-2.7 4.54 15.87 15.87 0 01-3.96 3.4c-1.45.86-3 1.53-4.63 1.98 0 0 2.84-4.15 5.8-8.2h-.3a6.82 6.82 0 116.8-6.98zm16.35 0v1.13c-.13 1.42-.48 2.81-1.02 4.13a15.73 15.73 0 01-2.68 4.56 15.86 15.86 0 01-3.96 3.33 18.29 18.29 0 01-4.68 1.99s2.83-4.13 5.79-8.18h-.3A6.82 6.82 0 1130 6.68z"
                    />
                  </svg>
                </section>
              </a>
            </section>
          </section>
        </section>
      </section>
    </section>
  </section>
  <section
    class="publication"
  >
    <section
      class="publication__columns"
    >
      <section
        class="publication__columns__main"
      >
        <h5>
          Amyloid precursor protein dimerization and synaptogenic function depend on copper binding to the growth factor-like domain.
        </h5>
        <section
          class="publication__authors"
        >
          <a
            href="/"
          >
            Baumkotter F.
          </a>
          , 
          <a
            href="/"
          >
            Schmidt N.
          </a>
          , 
          <a
            href="/"
          >
            Vargas C.
          </a>
          , 
          <a
            href="/"
          >
            Schilling S.
          </a>
          , 
          <a
            href="/"
          >
            Weber R.
          </a>
          , 
          <a
            href="/"
          >
            Wagner K.
          </a>
          , 
          <a
            href="/"
          >
            Fiedler S.
          </a>
          , 
          <a
            href="/"
          >
            Klug W.
          </a>
          , 
          <a
            href="/"
          >
            Radzimanowski J.
          </a>
          <button
            class="button tertiary"
            type="button"
          >
            [...]
          </button>
          , 
          <a
            href="/"
          >
            Kins S.
          </a>
        </section>
      </section>
      <section
        class="publication__columns__side"
      >
        <section
          class="publication__columns__side__item"
        >
          <svg
            fill="currentColor"
            height="2em"
            viewBox="0 0 30 32"
            width="1.875em"
          >
            <path
              d="M26.5 13.5h-8v8h8v-8z"
            />
            <path
              d="M26 14v7h-7v-7h7zm1-1h-9v9h9v-9z"
            />
            <path
              d="M29 1v28.06H17.05a3 3 0 00-2 .74V1.94a1 1 0 011-.94H29zm1-1H16.05a2 2 0 00-2 1.94V32h1a1.9 1.9 0 01.59-1.38 2 2 0 011.4-.56H30V0z"
            />
            <path
              d="M13.12 1a.91.91 0 01.88.94v27.78a2.78 2.78 0 00-1.81-.66H1V1h12.12zm0-1H0v30.06h12.19a1.84 1.84 0 011.32.56 1.93 1.93 0 01.55 1.38H15V1.94A1.91 1.91 0 0013.12 0z"
            />
            <path
              d="M22 5.5h5m-24 8h9m-9-4h7m-7-4h5m10 4h9m-24 8h9m-9 4h9m-9 4h9m6 0h9"
              stroke="currentColor"
              stroke-miterlimit="10"
            />
          </svg>
          <ul
            class="no-bullet"
          >
            <li>
              <a
                href="//www.ncbi.nlm.nih.gov/pubmed/25122912"
                rel="noopener noreferrer"
                target="_blank"
              >
                PubMed
              </a>
            </li>
            <li>
              <a
                href="//europepmc.org/article/MED/25122912"
                rel="noopener noreferrer"
                target="_blank"
              >
                Europe PMC
              </a>
            </li>
            <li>
              <small>
                <a
                  href="//dx.doi.org/10.1523/jneurosci.0180-14.2014"
                >
                  J. Neurosci. 34:11159-11172(2014)
                </a>
              </small>
            </li>
          </ul>
        </section>
        <section
          class="publication__columns__side__item"
        >
          <section
            class="publication__statistics"
          >
            <section
              class="publication__statistics__item"
            >
              <a
                href="/"
              >
                <section>
                  <small>
                    Cited in
                  </small>
                </section>
                <section
                  class="publication__statistics__bubble"
                >
                  <span
                    class="bubble--small colour-pastel-blue"
                  >
                    1
                  </span>
                  <svg
                    fill="currentColor"
                    height="15"
                    viewBox="0 0 30 22"
                    width="15"
                  >
                    <path
                      d="M13.65 6.68V7.7a.14.14 0 010 .11c-.13 1.42-.47 2.81-1.01 4.13a15.73 15.73 0 01-2.7 4.54 15.87 15.87 0 01-3.96 3.4c-1.45.86-3 1.53-4.63 1.98 0 0 2.84-4.15 5.8-8.2h-.3a6.82 6.82 0 116.8-6.98zm16.35 0v1.13c-.13 1.42-.48 2.81-1.02 4.13a15.73 15.73 0 01-2.68 4.56 15.86 15.86 0 01-3.96 3.33 18.29 18.29 0 01-4.68 1.99s2.83-4.13 5.79-8.18h-.3A6.82 6.82 0 1130 6.68z"
                    />
                  </svg>
                </section>
              </a>
            </section>
          </section>
        </section>
      </section>
    </section>
  </section>
</DocumentFragment>
`;
